app/src/main/java/org/schabi/newpipe/util/ShareUtils.java

```diff
@@ -140,33 +148,45 @@ private static void openInDefaultApp(final Context context, final Intent viewInt
     /**
      * Get the default app package name.
      * <p>
-     * If no app is set as default, it will return "android".
+     * If no app is set as default, it will return "android" (not on some devices because some
+     * OEMs changed the app chooser).
      * <p>
-     * Note: it doesn't return "android" on some devices because some OEMs changed the app chooser.
+     * If no app is installed on user's device to handle the intent, it will return null.
      *
      * @param context the context to use
      * @param intent  the intent to get default app
-     * @return the package name of the default app, or the app chooser if there's no default
+     * @return the package name of the default app, null if there is no app installed to handle the
+     * intent or the app chooser if there's no default
      */
     private static String getDefaultAppPackageName(final Context context, final Intent intent) {
-        return context.getPackageManager().resolveActivity(intent,
-                PackageManager.MATCH_DEFAULT_ONLY).activityInfo.packageName;
+        try {
+            return context.getPackageManager().resolveActivity(intent,
+                    PackageManager.MATCH_DEFAULT_ONLY).activityInfo.packageName;
+        } catch (final NullPointerException e) {
+            return null;
```

Comment:
If you return `null` here, please make sure to check the method's usages and add null checks to not create NPE there.
Also please add the `@Nullable` annotation to the method, so the IDE can warn when using this method and not checking for `null`.